### DISCUSSION TITLE
CRM Integration Issue/Dilemma.

### DISCUSSION THREAD

**Discussion by archoda** [solspace/craft-freeform#317] [Feature Requests]
For a site requiring multi-environment/instances (Local Dev, Staging, Production, etc…) it would be great if there was a simple way to enable/disable CRM implementation with environment variables. Maybe even a simple admin panel toggle switch at the least so as to not require a developer to have to remember to delete the instances implementation and/or manually mangle the service providers API settings to force failed requests.

A current client utilizes Hubspot and I’ve also noticed that even during testing where all caches are cleared (DNS, Browser, CraftCMS) and the Hubspot API Key was intentionally mangled with extra characters to try to prevent data transfer, Hubspot still seemed to receive and act on some data even while the return of the Ajax Fetch/Submit err’d as as expected with an “Invalid API”.

This would be a great help in the testing required for implementation, maintenance, updates, etc…

**Comment by kjmartens** [comment]
Thanks for your feedback, @archoda. This is an area of Freeform we do plan to improve in the future. It's not something we can address right now though.

**Comment by kjmartens** [comment]
The form builder and integrations have been overhauled in Freeform 5 for Craft 4 (now in beta). You can now set up multiple copies of an integration, and enable/disable them on the form inside the form builder. 🙂
